python/GafferImageUI/RGBAChannelsPlugValueWidget.py

```diff
@@ -100,14 +102,28 @@ def __menuDefinition( self ) :
 				text = prefix + "RGBA"
 				value = [ prefix + x for x in [ "R", "G", "B", "A" ] ]
 			else :
+				nonStandardLayers.add( text.split( "." )[0] )	
```

Comment:
If the first channel is non-standard, I get this :

```
UnboundLocalError: local variable 'text' referenced before assignment
```

Presumably even if it isn't, I'm still getting the text from the previous iteration? Also, `text.split( "." )[0]` is _not_ the layer name - layer's can be nested arbitrarily deep (e.g. left.diffuse for a stereo image). I think using `nonStandardLayers.add( GafferImage.ImageAlgo.layerName( channelName ) )` would do the trick...